### ISSUE TITLE
Use MappingQ1 when possible

### ISSUE THREAD

**Issue by bangerth** [geodynamics/aspect#999]

**Comment by bangerth** [comment]
By way of explanation, @ian-r-rose will add code via #965 that replaces the mapping object to be a unique pointer that will be initialized in the following way or similar:
```
//The mapping is given by a degree four MappingQ for the case
 +    //of a curved mesh, and a degree one MappingQ for a rectangular mesh.
 +    //If a free surface is enabled, this is swapped out for a MappingQ1Eulerian,
 +    //which allows for mesh deformation.
 +    mapping.reset( new MappingQ<dim>( geometry_model->has_curved_elements() ? 4 : 1,
 +                                      geometry_model->has_curved_elements() ? true : false ) );
```
This could be done in a better way using the (cheaper) `MappingQ1` class as follows:
```
//The mapping is given by a degree four MappingQ for the case
 +    //of a curved mesh, and a degree one MappingQ for a rectangular mesh.
 +    //If a free surface is enabled, this is swapped out for a MappingQ1Eulerian,
 +    //which allows for mesh deformation.
 +    mapping.reset( geometry_model->has_curved_elements() ?
                                new MappingQ<dim>(4, true) :
                                new MappingQ1<dim>());
```

**Comment by ian-r-rose** [comment]
I've got this one.

**Comment by gassmoeller** [comment]
Closed by #1041
